server/src/utils/mixerConnections/AtemConnection.ts

```diff
@@ -0,0 +1,415 @@
+        ]
+    }
+
+    private setPropsFromChannel(
+        channelIndex: number,
+        channel: IChannel,
+        fader: IFader,
+        source: FairlightAudioSource
+    ) {
+        if ((source.properties.gain + 1200) / 1800 !== fader.inputGain) {
+            // update gain
+            this.setFaderGain(
+                channelIndex,
+                (source.properties.gain + 1200) / 1800
+            )
+        }
+        if (!channel.fadeActive) {
+            if (
+                source.properties.mixOption === FairlightAudioMixOption.On &&
+                source.properties.faderGain > -10000
+            ) {
+                if (fader.muteOn) {
+                    // set mute off
+                    this.setMute(channelIndex, false)
+                }
+                if (!fader.pgmOn && !fader.voOn) {
+                    // set fader on
+                    this.setFaderPgm(channelIndex, true)
+                }
+            }
+            if (
+                source.properties.mixOption === FairlightAudioMixOption.Off &&
+                (fader.pgmOn || fader.voOn)
+            ) {
+                // set fader off
+                this.setFaderPgm(channelIndex, false)
+                this.setFaderVo(channelIndex, false)
+            }
+            if (
+                dbToFloat(source.properties.faderGain / 100) !==
+                fader.faderLevel
+            ) {
+                // set fader level
+                this.setFaderLevel(
+                    channelIndex,
+                    dbToFloat(source.properties.faderGain / 100)
+                )
+            }
+        }
+    }
+
+    private setFaderGain(
+        faderIndex: number,
+        gain: number,
+        update = true
+    ): void {
+        store.dispatch(storeInputGain(faderIndex, gain))
+        if (update) global.mainThreadHandler.updatePartialStore(faderIndex)
+    }
+    private setMute(faderIndex: number, muteOn: boolean, update = true): void {
+        store.dispatch(storeSetMute(faderIndex, muteOn))
+        if (update) global.mainThreadHandler.updatePartialStore(faderIndex)
+    }
+    private setFaderPgm(
+        faderIndex: number,
+        pgmOn: boolean,
+        update = true
+    ): void {
+        store.dispatch(storeSetPgm(faderIndex, pgmOn))
+        if (update) global.mainThreadHandler.updatePartialStore(faderIndex)
+    }
+    private setFaderVo(faderIndex: number, voOn: boolean, update = true): void {
+        store.dispatch(storeSetVo(faderIndex, voOn))
+        if (update) global.mainThreadHandler.updatePartialStore(faderIndex)
+    }
+    private setFaderLevel(
+        faderIndex: number,
+        level: number,
+        update = true
+    ): void {
+        store.dispatch(storeFaderLevel(faderIndex, level))
+        if (update) global.mainThreadHandler.updatePartialStore(faderIndex)
+    }
+    private setChannelLabel(
+        channelIndex: number,
+        label: string,
+        update = true
```

Comment:
A minor thing: Would be nice with parameter types, even for parameters with default values.